tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/dialogs/UIWorkingSetWizardsAuto.java

```diff
@@ -135,10 +137,18 @@ private void removeAllWorkingSets() {
 
 	private void cleanupWorkspace() {
 		try {
+			PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().closeAllEditors(false);
+			Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_BUILD, null);
 			ResourcesPlugin.getWorkspace().getRoot().delete(true, null);
 		} catch (CoreException e) {
 			TestPlugin.getDefault().getLog().log(e.getStatus());
 			throw createAssertionError(e);
+		} catch (OperationCanceledException e) {
+			e.printStackTrace();
+			throw new AssertionError(e.getMessage(), e);
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+			throw new AssertionError(e.getMessage(), e);
```

Comment:
I would propose to have the method simply throw the exception, either directly or after doing some logging. So you could just replace the catch blocks with a `throws` declaration on the method signature. The stack trace should be reported then anyway (and otherwise I would not recommend to simply print it but to use some logger for logging it).

Reply:
Addressed! https://github.com/eclipse-platform/eclipse.platform.ui/commit/ded182da305777b5c3fb5cfbedd88cb02449cb1f